Fix RPC semantic conventions (#386)
open-telemetry/opentelemetry-specification: specification/data-rpc.md

| Attribute name | Notes and examples                                           | Required? |
| -------------- | ------------------------------------------------------------ | --------- |
| `component`    | Declares that this is a grpc component. Value MUST be `"grpc"`. | Yes       |

> | `rpc.service`  | The service name, must be equal to the $service part in the span name. | Yes |

Reviewer: Why do we need to duplicate this info?

Author: The service name is one of the most interesting attributes for grouping and filtering on the backend so I would make that one available without having to parse the span name. I deem the attributes defined here more stable and reliable than the span name and would prefer those for analyzing.
This is, however, not part of my changeset but was already there before my PR.

Reviewer 2: It was renamed from `peer.name` in #349.

> | `net.peer.port`  | See [network attributes][]. | Yes |

Reviewer: I am confused between this and the description of the PR. As you mentioned in the description the port is not available on the server side, so why is this required then?

Author: The hostname of the peer would usually not be available but the remote port should be there.

Reviewer: I still am confused, when you open a socket to a remote location usually it get allocated a port that is not easy to associate with the application, but I think you always have the ip of the peer.

Author: `net.peer.*` describes the remote side of the connection. As a client, this would be the server I'm connecting to; as a server, this would be the client connecting to me. In both cases I should be able to know both the address as well as the port that I'm connecting to, otherwise I wouldn't be able to send a message or respond.

In GRPC for Java, for example, the client knows the server's address and port since it has to be provided to [ManagedChannelBuilder.forAddress(host, port)](https://grpc.github.io/grpc-java/javadoc/io/grpc/ManagedChannelBuilder.html#forAddress-java.lang.String-int-) and the server can access the same for the client via [serverCall.getAttributes().get(Grpc.TRANSPORT_ATTR_REMOTE_ADDR)](https://grpc.github.io/grpc-java/javadoc/io/grpc/Grpc.html#TRANSPORT_ATTR_REMOTE_ADDR), which returns a `java.net.InetSocketAddress` containing the client's `InetAddress` and port.

Reviewer: but the port on the client is useless, and changes between connections from same peers. Why is this required in that case?

Author: It's not useless if you have infrastructure monitoring that tracks open TCP connections on a host when the process itself is not instrumented.

Author: updated as discussed in the SIG meeting

| Attribute name | Notes and examples                                           | Required? |
| -------------- | ------------------------------------------------------------ | --------- |
| `component`    | Declares that this is a grpc component. Value MUST be `"grpc"`. | Yes       |
| `rpc.service`  | The service name, must be equal to the $service part in the span name. | Yes |
| `net.peer.ip`  | See [network attributes][]. | See below |
| `net.peer.name`  | See [network attributes][]. | See below |
| `net.peer.port`  | See [network attributes][]. | See below |

At least one of [network attributes][] `net.peer.name` or `net.peer.ip` is required.

> For client-side spans describing the server they are connecting to, `net.peer.port` is required.

Reviewer 2: ```suggestion
For client-side spans `net.peer.port` is required (it describes the server port they are connecting to).
```

> For server-side spans describing the client connecting to them, `net.peer.port` is optional.

Reviewer 2: ```suggestion
For server-side spans `net.peer.port` is optional (it describes the port the client is connecting from).
```

| -------------- | ------------------------------------------------------------ | --------- |
| `component`    | Declares that this is a grpc component. Value MUST be `"grpc"` | Yes       |
| `rpc.service`  | The service name, must be equal to the $service part in the span name. | Yes |

> | `net.peer.port`  | see [network attributes][] | Yes |

Reviewer 2: ```suggestion
| `net.peer.port`  | See [network attributes][] | Yes |
```

Additionally, at least one of `net.peer.name` or `net.peer.ip` is required.

> These attributes are also defined in [network attributes][].

Reviewer 2: Why "also"? As far as I can see, they are *only* defined there.

Author: Also, since the `net.peer.port` mentioned in the line above is also defined there.

Author: I rephrased it :)

| `component`    | Declares that this is a grpc component. Value MUST be `"grpc"` | Yes       |
| `rpc.service`  | The service name, must be equal to the $service part in the span name. | Yes |

Additionally, at least one of `net.peer.name` or `net.peer.ip` is required as well as `net.peer.port`.

> These are defined in [network attributes][].

Reviewer 3: nit: Could you just add a line for `net.peer.port` to the table above, then remove "as well as `net.peer.port`"?

Author: sure, done!
